client/src/util/fetchFromAPI.test.js

```diff
@@ -6,13 +6,13 @@ beforeEach(() => {
 });
 
 it("mocks the fetch request", async () => {
-  fetch.mockResponseOnce();
+  fetch.mockResponseOnce(JSON.stringify({ foodMenu: { title: "potato" } }));
   const response = await queryAPI({
     url: "/",
     onSuccess: jest.fn(),
     onError: jest.fn(),
   });
-  expect(response).toEqual({});
+  expect(response).toEqual({ title: "potato" });
```

Comment:
What you have here would be exactly right _if_ `queryAPI` returned a response.  Unfortunately, the way I had you design it, it doesn't return a response.  Instead, it uses callbacks for success and error states.  This isn't the first time that design has caused you friction and I'm pretty sure the right thing to do is to redesign the `queryAPI` function instead of constantly fighting it.

Thoughts?

Reply:
Yes I'm fine redesigning it - I have created a new branch for it ...but based on your comments to close this PR is it safe to assume passing the callback states will validate the tests?

Reply:
I would keep this PR open, for the record.

The tests could be written to validate that the callbacks are called.  You might even be able to validate that they are called with a particular payload.

But I think it's better to redesign this instead of focusing on how to test my bad design.

Reply:
Ok copy that

Reply:
- [ ] Remove onSuccess callback
- [ ] return the result object
- [ ] Have the onError default behavior be a toast containing the error message
- [ ] Test everything